packages/x-data-grid-pro/src/hooks/features/dataSource/utils.ts

```diff
@@ -0,0 +1,103 @@
+import { GridRowId } from '@mui/x-data-grid';
+import { GridPrivateApiPro } from '../../../models/gridApiPro';
+
+const MAX_CONCURRENT_REQUESTS = Infinity;
+
+export const runIfServerMode = (modeProp: 'server' | 'client', fn: Function) => () => {
+  if (modeProp === 'server') {
+    fn();
+  }
+};
+
+export enum RequestStatus {
+  QUEUED,
+  PENDING,
+  SETTLED,
+  UNKNOWN,
+}
+
+/**
+ * Fetches row children from the server with option to limit the number of concurrent requests
+ * Determines the status of a request based on the enum `RequestStatus`
+ * Uses `GridRowId` to uniquely identify a request
+ */
+export class NestedDataManager {
```

Comment:
Why `NestedDataManager` and not `DataManager`? It would shorten a few lines and make the code less verbose if the name was shortened.

Reply:
Mainly because it only manages nested data. I can rename it if you have a hard preference. To me, `nested` word makes the name more self-explanatory.